### ISSUE TITLE
Object-Centric Image Generation from Layouts

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#115]
*AAAI 2021*

# TL;DR

- ***Scene Graph Similarity Module***: Generate Scene Graph from input Layout and computes similarity with synthesized image.
- ***Instance-aware Conditioning***: Provide bbox instance boundary (obtained from bbox directly) for spatially adaptive normalization parameter estimation

# Motivation

- Shortcomings of previous L2I methods: (1) Low layout fidelity (2) Merging objects (unclear boundary) (3) Spurious Objects

# Previous Work

- Layout2im
- LostGAN
- LostGAN-v2

# Method

## Overview
![image](https://user-images.githubusercontent.com/46308407/130729740-b88b85ef-522a-41b0-8d29-77324dfddbd7.png)
- From a layout, a scene graph as well as image is generated.
- SGSM enforces Generator output to match scene graph obtained from layout (*easier, simpler to distil relational information*)

## Scene Graph Similarity Module
![image](https://user-images.githubusercontent.com/46308407/130730001-59aa1d3c-a3ca-4284-8da2-8d40a34739fd.png)
- Extract Local/Global visual/graph features by convolution/graph convolution.
- Compute *visually attended local graph features*
![image](https://user-images.githubusercontent.com/46308407/130730327-283f01f7-476b-4e42-a1a9-0b008b28f19e.png)
- Compute similarity between image and scene graph both in local/global scope.
![image](https://user-images.githubusercontent.com/46308407/130730363-e21fc980-32de-4f8d-911c-01275653f1cf.png)
- Define a probability model based on similarity scores
![image](https://user-images.githubusercontent.com/46308407/130730389-005262a0-27d0-4129-a36c-8affa0162a7a.png)
![image](https://user-images.githubusercontent.com/46308407/130730397-226ecc60-df4b-4bc9-81c6-3a71f029bf77.png)
- Negative samples are selected by ***mis-matching*** image and scene graph pair in the mini-batch.
- *Personally think it's similar to DAMSM in AttnGAN (text-to-image task)

## Instance-aware Conditioning
- Previous methods provide near-binary masks for spatially adaptive normalization. This cannot address ***overlapping objects***.
- Provide object ***boundary*** extracted directly from bboxes (no additional information) on top of embeddings and bboxes as a ***prior*** to encourage model to generate distinct objects.
![image](https://user-images.githubusercontent.com/46308407/130730633-897cfe32-2d7f-43f7-ad36-a03830a6f126.png)

## Adversarial Training
- Res-Net generator as in *LostGAN*.
- Object Discriminator + 2 multi-scale Image Discriminator (original resolution & downsampled)

## Objective
![image](https://user-images.githubusercontent.com/46308407/130730786-2294b350-ad8b-42eb-941b-7dfd919cfb37.png)
![image](https://user-images.githubusercontent.com/46308407/130730797-6596d703-6f36-4f46-86de-47535d13ac78.png)
- Lp refers to VGG-19 perceptual loss between original image and synthesized output.

# Experiments

## Dataset
- Visual Genome
- COCO

## Metric
- Inception Score (*legacy metric*)
- FID
- Classification Accuracy
- ***Scene FID***: crop objects and upsample them to 224 x 224 then compute FID (FID: single manifold assumption)

## Baseline
- Layout-to-Image: Layout2Im, LostGAN, LostGAN-v2
- Scene Graph-to-Image: sg2im, SOARISG
- Semantic Synthesis: SPADE, Pix2Pix-HD

![image](https://user-images.githubusercontent.com/46308407/130731622-e46ea433-a8c5-4a3a-bae3-833a1b8143c5.png)
![image](https://user-images.githubusercontent.com/46308407/130731648-05a378dc-587f-44df-8291-cdcd90a6814a.png)
![image](https://user-images.githubusercontent.com/46308407/130731668-26518c3c-b175-43d5-a024-93bbe8854c41.png)
![image](https://user-images.githubusercontent.com/46308407/130731677-b775689f-5b3c-40ee-9d99-e6a956f6d632.png)

## Ablation
![image](https://user-images.githubusercontent.com/46308407/130731787-7cc4f8c2-536b-49c8-93e0-77b140a812aa.png)

# Comments
- Looking at the ablation, additional boundary information contributes to only ***marginal gain*** in quantitative metric, while SGSM is ***crucial*** factor behind their decent performance.
- Wonder adding scene graph matching helped performance in terms of whether *additional information* or *easier to distil information query*. Thought bbox would be an easier que for the model to reason spatial relationships.
- Perhaps scene graph and layouts are complimentary information, best employed when provided together. 
- Scene graph has some ambiguities in terms of spatial relationship, while bounding boxes are less straight-forward to convey the information to the model. Also, scene graphs can be more descriptive as they can represent relationships other than spatial positions (ex: holding, walking, facing, etc)
